otx/algorithms/segmentation/adapters/mmseg/task.py

```diff
@@ -0,0 +1,563 @@
+            data_classes = [label.name for label in self._labels]
+        else:
+            data_classes = None
+        model_classes = [label.name for label in self._model_label_schema]
+
+        recipe_cfg.work_dir = self._output_path
+        recipe_cfg.resume = self._resume
+
+        if self._train_type == TrainType.Incremental:
+            configurer = IncrSegmentationConfigurer()
+        elif self._train_type == TrainType.Semisupervised:
+            configurer = SemiSLSegmentationConfigurer()
+        else:
+            configurer = SegmentationConfigurer()
+        cfg = configurer.configure(
+            recipe_cfg, self._model_ckpt, data_cfg, training, subset, ir_options, data_classes, model_classes
+        )
+        self._config = cfg
+        return cfg
+
+    def build_model(
+        self,
+        cfg: Config,
+        fp16: bool = False,
+        **kwargs,
+    ) -> torch.nn.Module:
+        """Build model from model_builder."""
+        model_builder = getattr(self, "model_builder", build_segmentor)
+        model = model_builder(cfg, **kwargs)
+        if bool(fp16):
+            wrap_fp16_model(model)
+        return model
+
+    def _infer_model(
+        self,
+        dataset: DatasetEntity,
+        inference_parameters: Optional[InferenceParameters] = None,
+    ):
+        """Main infer function."""
+        self._data_cfg = ConfigDict(
+            data=ConfigDict(
+                train=ConfigDict(
+                    otx_dataset=None,
+                    labels=self._labels,
+                ),
+                test=ConfigDict(
+                    otx_dataset=dataset,
+                    labels=self._labels,
+                ),
+            )
+        )
+
+        dump_features = True
+
+        self._init_task()
+
+        cfg = self.configure(False, "test", None)
+        logger.info("infer!")
+
+        # Data loader
+        mm_dataset = build_dataset(cfg.data.test)
+        dataloader = build_dataloader(
+            mm_dataset,
+            samples_per_gpu=cfg.data.get("samples_per_gpu", 1),
+            workers_per_gpu=cfg.data.get("workers_per_gpu", 0),
+            num_gpus=len(cfg.gpu_ids),
+            dist=cfg.distributed,
+            seed=cfg.get("seed", None),
+            persistent_workers=False,
+            shuffle=False,
+        )
+
+        # Target classes
+        if "task_adapt" in cfg:
+            target_classes = cfg.task_adapt.final
+            if len(target_classes) < 1:
+                raise KeyError(
+                    f"target_classes={target_classes} is empty check the metadata from model ckpt or recipe "
+                    "configuration"
+                )
+        else:
+            target_classes = mm_dataset.CLASSES
+
+        # Model
+        model = self.build_model(cfg, fp16=cfg.get("fp16", False))
+        model.CLASSES = target_classes
+        model.eval()
+        feature_model = model.model_s if self._train_type == TrainType.Semisupervised else model
```

Comment:
I can not find any clue for this in the previous branch. Could you explain why feature_model is model.model_s not model?

Reply:
Please refer `MeanTeacherSegmentor` on here : https://github.com/openvinotoolkit/training_extensions/blob/develop/otx/algorithms/segmentation/adapters/mmseg/models/segmentors/mean_teacher_segmentor.py#L41-L59

For semi-sl on semantic segmentation, `model.model_s` should be used as `feature_model`.

Reply:
Got it, so previous implementation was wrong, wasn't it?

Reply:
Yes, I believe that. I've checked `model.model_s` outputs feature vector well.